### PR TITLE
Refactor how we get etcd endpoints in CAPI clusters

### PR DESCRIPTION
**Pull request by weseven**
This fixes https://github.com/giantswarm/giantswarm/issues/29017, https://github.com/giantswarm/giantswarm/issues/28849, probably others.

On CAPI clusters etcd-backup-operator uses the apiserver as proxy to the etcd pods in the cluster. 
This PR changes how we get those pods: the current implementation was getting a list of controlplane nodes and then building the pod name as "etcd-$nodename", but the filter part was not working as expected and in the clusters where the first node on the list is not a controlplane one it caused backups to fail.
With these changes, we get the pods that have the label "component=etcd", which should be more robust.

Huge thanks and kudos to @Gacko for the troubleshooting help!

### CODE REVIEW

File: pkg/giantnetes/utils.go [hunk 1]
```diff
@@ -307,18 +305,21 @@ func (u *Utils) getEtcdEndpoint(ctx context.Context, cluster Cluster) (string, e
 				return "", microerror.Maskf(executionFailedError, "error creating CAPI cluster kubernetes client for cluster  %#q with error %#q", cluster.clusterKey.Name, err)
 			}
 
-			err = targetCtrlClient.List(ctx, &nodeList, ctrl.HasLabels{LabelCAPIControlPlaneNode})
+			// Specify the label for etcd pods
+			labelSelector := client.MatchingLabels(map[string]string{EtcdLabelComponent: EtcdLabelValue})
```
Comment (Gacko) [inline pkg/giantnetes/utils.go:309]: I remember there were other labels, too. Something like `tier: control-plane`. Would it make sense to also add this to have this part more precise?

**Comment by Gacko** [comment]
cc @calvix

File: pkg/giantnetes/utils.go (hunk 1 above)
Comment (weseven) [reply pkg/giantnetes/utils.go]: (added that additional label in https://github.com/giantswarm/etcd-backup-operator/pull/438/commits/1cc5fd16469eda0469b09f3fa5c86a8bfcebf1ee)

**Review by Gacko — APPROVED** [review]

**Review by njuettner — APPROVED** [review]
Nice 🚀

**Pull request merged**